### PR TITLE
set obj.common.def as initial value

### PR DESCRIPTION
**Pull request by foxriver76**
-on setObject
-and setObjectNotExists
-solves #228 

sorry, some lines have been formatted accidentally

### CODE REVIEW

File: lib/adapter.js [hunk 1]
```diff
@@ -1137,8 +1139,13 @@ function Adapter(options) {
                     if (!obj.common.hasOwnProperty('type')) {
                         logger.warn(that.namespace + ' setObject ' + id + ' (type=' + obj.type + ') property common.type missing!');
                     }
+                    if (obj.common.hasOwnProperty('def') && obj.common.def !== undefined) {
```
Comment (AlCalzone) [inline lib/adapter.js]: This could just be `if (obj.common.def != undefined)`. `!=` checks for `null` and `undefined` as opposed to `!==`

Comment (foxriver76) [reply lib/adapter.js]: Thanks, will adjust it this evening.

Comment (AlCalzone) [reply lib/adapter.js]: Although... not sure if `null` is a valid default value? Then it should just be `!== undefined`

**Comment by Apollon77** [comment]
!== :-)

**Comment by Apollon77** [comment]
In fact should be same as https://github.com/ioBroker/ioBroker.js-controller/blob/master/lib/adapter.js#L2236 (because already existing and so defines the "should be that way" :-)

**Comment by Apollon77** [comment]
Looks good!

**Comment by AlCalzone** [comment]
👍

**Comment by Apollon77** [comment]
seems to be ok now, I will squash for the merge!

**Pull request merged**